### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by yahyaelganyni1**
this pull request is for unit testing and jest snapshot

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hello Team,

You've done pretty well so far!
But for a few issues, you are almost done with this. Hang in there!


## Required Changes ♻️
_Please check the comments under the review._


Happy coding, Cheers! 🚀🚀

Do feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether it is valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__test__/calculator.test.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
+import calculator from '../components/logic/calculate';
```
Comment (mikenath223) [inline src/__test__/calculator.test.js:1]: - [ ] Please use the react testing library to simulate user interaction, by firing click events.

File: src/__test__/calculator.test.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+  test('1 % 1 = 0', () => {
+    let data = {};
+
+    data = { ...data, ...calculator(data, '1') };
+    data = { ...data, ...calculator(data, '%') };
+    data = { ...data, ...calculator(data, '1') };
+    data = { ...data, ...calculator(data, '=') };
+
+    expect(data).toEqual({
```
Comment (mikenath223) [inline src/__test__/calculator.test.js:71]: - [ ] Please test edge cases like division by zero.

File: src/__test__/calculator.test.js (hunk 1 above)
Comment (ahmed-alfarouq) [reply src/__test__/calculator.test.js:1]: Hi @mikenath223,
I didn't get exactly what you mean. Can you tell me more details, please?
Thanks for the feedback

File: src/__test__/calculator.test.js (hunk 2 above)
Comment (ahmed-alfarouq) [reply src/__test__/calculator.test.js:71]: Hi @mikenath223,
We did that.
Thanks for the feedback, sir
